### ISSUE TITLE
Script sigterm handler does not complete

### ISSUE THREAD

**Issue by timdiels** [nextflow-io/nextflow#322]
## Use case
I have a script with a sigterm handler which needs to cancel a job and remove its result, but the handler does not always fully complete. One thing I've noticed is that it does not wait for blocking commands. I've boiled this down to the problem below.

## Actual
On nextflow receiving sigint, the process script receives sigterm twice (`hi` appears twice) and does not wait for the sleep in the signal handler (the run exits in less than 10 seconds and `sleep` appears in `ps` output).
```
[limyreth@petrov tmp_nextflow]$ ps                
  PID TTY          TIME CMD
 3579 pts/0    00:00:00 zsh
14355 pts/0    00:00:00 ps
[limyreth@petrov tmp_nextflow]$ nextflow run .
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on
N E X T F L O W  ~  version 0.24.0
Launching `./main.nf` [determined_golick] - revision: a0c336aa6a
[warm up] executor > local
[a4/b96a54] Submitted process > test (1)
^C
WARN: Killing pending tasks (1)
[limyreth@petrov tmp_nextflow]$ ps
  PID TTY          TIME CMD
 3579 pts/0    00:00:00 zsh
14427 pts/0    00:00:00 tee
14428 pts/0    00:00:00 tee
14429 pts/0    00:00:00 bash
14453 pts/0    00:00:00 sleep
14460 pts/0    00:00:00 ps
[limyreth@petrov tmp_nextflow]$ cat work/a4/b96a54120c7179bf3baa3ca48580f5/.command.out 
hi
hi
bye
[limyreth@petrov tmp_nextflow]$ cat work/a4/b96a54120c7179bf3baa3ca48580f5/.command.err
Terminated
[limyreth@petrov tmp_nextflow]$ cat work/a4/b96a54120c7179bf3baa3ca48580f5/.command.log
hi
Terminated
hi
bye
```

## Expected
On sigint, the process script ideally receives sigterm once, but the main thing is that it should wait for `sleep 10` to finish before echoing hi and exiting.
```
[limyreth@petrov tmp_nextflow]$ ps                
  PID TTY          TIME CMD
 3579 pts/0    00:00:00 zsh
14355 pts/0    00:00:00 ps
[limyreth@petrov tmp_nextflow]$ cat work/a4/b96a54120c7179bf3baa3ca48580f5/.command.out 
hi
bye
```

## Workaround
Putting the body of `cancel` in a separate script `cleanup.sh` and changing the trap to `trap 'setsid path/to/cleanup.sh' SIGTERM` does work.

**Comment by pditommaso** [comment]
I've uploaded a new snapshot that should solve this problem. You can try by using the following command: 

```
NXF_VER=0.24.3-SNAPSHOT nextflow run .. etc
```

**Comment by pditommaso** [comment]
Any feedback on this?

**Comment by timdiels** [comment]
I'll give it a spin this weekend

**Comment by pditommaso** [comment]
This may be related to #172

**Comment by pditommaso** [comment]
Released with version `0.24.3`.
